zebrad/src/components/rpc.rs

```diff
@@ -0,0 +1,37 @@
+//! An RPC endpoint.
+
+use jsonrpc_core;
+
+use jsonrpc_http_server::ServerBuilder;
+
+use zebra_rpc::rpc::{Rpc, RpcImpl};
+
+pub mod config;
+pub use config::Config;
+
+/// Zebra RPC Server
+pub struct RpcServer {}
+
+impl RpcServer {
+    /// Start a new RPC server endpoint
+    pub async fn new(config: Config) -> Self {
```

Comment:
Blocking Problem (2/3): Return the spawned task.

```suggestion
    /// Start a new RPC server endpoint
    pub fn spawn(config: Config) -> JoinHandle<()> {
```

Reply:
https://github.com/ZcashFoundation/zebra/pull/3589/commits/8ed2047734f5915cc227f861d5e82991137d5ca2